File: LeetCode-Problems/Easy/Middle-Of-Linked-List.java
//Java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode middleNode(ListNode head) {
        ListNode current = head;
        int leng = 0;
        while (current != null){
            leng++;
            current = current.next;
        }
        current = head;
        int x = 0;
        while (x < leng/2){
            current = current.next;
            x++;
        }
        return current;   
    }
}

File: LeetCode-Problems/Easy/Middle-Of-Linked-List.py
# Python
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def middleNode(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        current = head
        leng = 0
        while(current != None):
            leng = leng + 1
            current = current.next
        current = head
        x = 0
        while (x < leng/2):
            current = current.next
            x = x+1
        return current
